wayshot/src/wayshot.rs

```diff
@@ -99,192 +68,65 @@ fn main() -> Result<(), Box<dyn Error>> {
         file_path = Some(utils::get_default_file_name(extension));
     }
 
-    let mut conn = Connection::connect_to_env().unwrap();
-    let (mut globals, _) = registry_queue_init::<WayshotState>(&conn).unwrap();
+    let wayshot_conn = WayshotConnection::new()?;
 
     if args.get_flag("listoutputs") {
-        let valid_outputs = output::get_all_outputs(&mut globals, &mut conn);
+        let valid_outputs = wayshot_conn.get_all_outputs();
         for output in valid_outputs {
             log::info!("{:#?}", output.name);
         }
         exit(1);
     }
 
-    let mut cursor_overlay: i32 = 0;
+    let mut cursor_overlay = false;
     if args.get_flag("cursor") {
-        cursor_overlay = 1;
+        cursor_overlay = true;
     }
 
-    let capture_area = if let Some(slurpregion) = args.get_one::<String>("slurp") {
-        match utils::parse_geometry(slurpregion) {
-            Some(region) => (wl_output::Transform::Normal, region),
-            None => {
-                log::error!("Invalid geometry specification");
-                exit(1);
-            }
+    let image_buffer = if let Some(slurpregion) = args.get_one::<String>("slurp") {
+        if let Some(region) = utils::parse_geometry(slurpregion) {
+            wayshot_conn.screenshot((Transform::Normal, region), cursor_overlay)?
```

Comment:
We should not need to hardcode the normal transform anymore. That was mostly a hack because we were going to set the transform later anyways. It would be more sensible to have that be abstracted away by the API.

If we *have* to hardcode normal just as a workaround ( which should no longer exist, ideally speaking ), then we should atleast mention that in the doc comments for the function.